### PR TITLE
docs/BOOTING: add a few lines on the console enabling

### PR DESCRIPTION
**Pull request by rouming**
Compilation of kernel serial console and console access documentation. Just a few lines to help to setup console properly.

### CODE REVIEW

File: docs/BOOTING.md [hunk 1]
```diff
@@ -136,6 +136,23 @@ in grub.cfg with graphical GRUB menu to get the device to boot again.
...
    They will be added to kernel cmdline.
 
+As an example of using serial port and VGA as a console the following kernel arguments
+should be used:
+
+```
+set_global dom0_extra_args "$dom0_extra_args console=ttyS0,115200n8 console=tty0"
+```
+
+defines that opening ``/dev/console`` will get you the current foreground virtual
```
Comment (deitch) [inline docs/BOOTING.md:146]: I think you meant these to be single-ticks.

File: docs/BOOTING.md [hunk 2]
```diff
@@ -136,6 +136,23 @@ in grub.cfg with graphical GRUB menu to get the device to boot again.
...
+console, and kernel messages will appear on both the VGA console and the 1st serial
+port (ttyS0 or COM1) at 115200 baud.
```
Comment (deitch) [inline docs/BOOTING.md:148]: What is it that makes them appear on both the ttyS0 _and_ the VGA? Is it something in the config here? Or is that hard-wired in somewhere? Worth explaining.

File: docs/BOOTING.md [hunk 3]
```diff
@@ -136,6 +136,23 @@ in grub.cfg with graphical GRUB menu to get the device to boot again.
...
+
+Details about serial console kernel arguments can be also seen [here](https://www.kernel.org/doc/html/v6.1/admin-guide/serial-console.html).
```
Comment (deitch) [inline docs/BOOTING.md:150]: Maybe, "Details about serial console kernel arguments for Linux in general are available on official kernel documentation.", with the hyperlink on "official kernel documentation"?

File: docs/BOOTING.md [hunk 4]
```diff
@@ -136,6 +136,23 @@ in grub.cfg with graphical GRUB menu to get the device to boot again.
 3. General kernel parameters may be adjusted with `set_global dom0_extra_args "$dom0_extra_args OPTION1=VAL1 OPTION2 "`.
    They will be added to kernel cmdline.
 
+As an example of using serial port and VGA as a console the following kernel arguments
+should be used:
+
+```
+set_global dom0_extra_args "$dom0_extra_args console=ttyS0,115200n8 console=tty0"
```
Comment (rene) [inline docs/BOOTING.md:143]: Although set `dom0_extra_args` will work (because it's the last variable listed in the boot command line and the kernel will always consider the default console as the last one declared), we have a specific variable for console: `dom0_console`, so I think it's better to instruct users to change `dom0_console` instead of `dom0_extra_args`...

Comment (rouming) [reply docs/BOOTING.md:143]: frankly, I do not understand how it is used and what is the difference between `dom0_console` and `hv_console`.

Comment (rene) [reply docs/BOOTING.md:143]: At [line 431](https://github.com/lf-edge/eve/blob/master/pkg/grub/rootfs.cfg#L431) of the rootfs.cfg we have the definition of the main boot entry menu:

```
menuentry "Boot ${rootfs_title}${rootfs_title_suffix}" {
     $load_hv_cmd $hv $hv_console $hv_platform_tweaks $hv_dom0_mem_settings $hv_dom0_cpu_settings $hv_extra_args
     $load_dom0_cmd $dom0 $dom0_console $dom0_rootfs $hv_dom0_mem_settings $hv_dom0_cpu_settings $hv_eve_mem_settings $hv_eve_cpu_settings $hv_ctrd_mem_settings $hv_ctrd_cpu_settings $hv_watchdog_timer $dom0_platform_tweaks $dom0_flavor_tweaks $dom0_cmdline $dom0_extra_args
     do_if_args $load_devicetree_cmd $devicetree
     # ucode initrd image must be the first!
     do_if_args $load_initrd_cmd $ucode
     do_if_args $load_initrd_cmd $initrd
}
```

So we have two commands for boot: `load_hv_cmd` and `load_dom0_cmd`, as you can see, all `hv_*` variables will be used with the `load_hv_cmd` while all variables `dom0_*` will be used with `load_dom0_cmd`. The `load_hv_cmd` is only useful for Xen, where we will indeed define parameters for the hypervisor and for the dom0 (privileged guest). However, for KVM, if you look at [line 356](https://github.com/lf-edge/eve/blob/master/pkg/grub/rootfs.cfg#L356) it just defined as a true command, which means it will do literally nothing.... so for KVM only `dom0_*` variables are really valid...

So in summary:

For KVM:
- Only `dom0_console` is required to change console

For Xen:
- `dom0_console` is required to change only the console of dom0 (not Xen output)
- `hv_console` is required to change only Xen output

Comment (rouming) [reply docs/BOOTING.md:143]: thanks for the explanation, so basically the key here is actual GRUB commands, either "xen_hypervisor", either "linux", etc, and those commands names are prefixes for whole configuration line, which consists of `dom0_` or `hv_`.

Comment (rene) [reply docs/BOOTING.md]: exactly @rouming ...

File: docs/BOOTING.md [hunk 5]
```diff
@@ -136,6 +136,24 @@ in grub.cfg with graphical GRUB menu to get the device to boot again.
 3. General kernel parameters may be adjusted with `set_global dom0_extra_args "$dom0_extra_args OPTION1=VAL1 OPTION2 "`.
    They will be added to kernel cmdline.
 
+As an example of using serial port and VGA as a console for the Dom0 (in Xen terminology,
+but also applies for KVM, which means the "host" operating system) the following
+arguments should be used:
+
+```
```
Comment (rene) [inline docs/BOOTING.md:143]: use ` ```sh ` to fix the yetus error (although it's not a proper shell code, but at least it should coloring the variables)

File: docs/BOOTING.md (hunk 2 above)
Comment (rouming) [reply docs/BOOTING.md:149]: Done

File: docs/BOOTING.md (hunk 1 above)
Comment (rouming) [reply docs/BOOTING.md:147]: Done

File: docs/BOOTING.md (hunk 3 above)
Comment (rouming) [reply docs/BOOTING.md:151]: Done

**Review by deitch — APPROVED** [review]
LGTM.
